biggraphite/drivers/elasticsearch.py

```diff
@@ -399,21 +472,78 @@ def delete_metric(self, name):
     def delete_directory(self, name):
         self._directory_names.remove(name)
 
-    @staticmethod
-    def __glob_names(names, glob):
-        results = bg_glob.glob(names, glob)
-        results.sort()
-        return results
+    def _search_metrics_from_glob(self, glob):
+        search = elasticsearch_dsl.Search()
+        search = search.using(self.client).index("%s*" % self._index_prefix).source('name')
+
+        components = self.__glob_parser.parse(glob)
+
+        # Handle glob with globstar(s).
+        globstars = components.count(bg_glob.Globstar())
+        if globstars:
+            name_regexp = "\\.".join([_parse_regexp_component(c) for c in components])
+            return True, search.filter('regexp', **{"name": name_regexp})
+
+        # Handle fully defined glob.
```

Comment:
There is another optimization that can be done here when there is something that looks like  'a.b.c.*.*.*': we can do name: 'a.b.c.*' instead of searching individual components.

Add a TODO for this.

Reply:
It's already implemented, right ?

Reply:
TODO added